internal/datatransfer/monitor.go

```diff
@@ -39,6 +39,14 @@ func (m *Monitor) Start() {
 	}()
 }
 
+func (m *Monitor) Deregister() error {
+	log.Info("Stopping monitor ticker")
+	if m.ticker != nil {
+		m.ticker.Stop()
+	}
+	return nil
+}
```

Comment:
Can you add to the Start() function a message that the monitor is stopped when the for loop is exited ?